src/easylink/implementation.py

```diff
@@ -111,15 +116,82 @@ def _validate_expected_steps(self, logs: list[str]) -> list[str]:
             )
         return logs
 
-    def _validate_container_exists(
+    def _download_and_validate_image(
```

Comment:
this is kinda hacky to stick the downloading w/in the Implementation validation method but it seemed easiest.